driver/src/main/java/org/neo4j/driver/internal/net/SocketClient.java

```diff
@@ -120,7 +119,10 @@ public void start()
         try
         {
             logger.debug( "~~ [CONNECT] %s", address );
-            setChannel( ChannelFactory.create( address, securityPlan, timeoutMillis, logger ) );
+            if( channel == null )
```

Comment:
The check here is to allow injection of a mocked channel into this class for testing. This null check is for `do not create a new channel if I already give you a fake channel`.